Approved/enhanced-array-literals.md

```diff
@@ -0,0 +1,209 @@
+    // This is a compilation error.
+    let empty3 = [];
+    Message($"Empty array: {empty3}");
+
+    // In cases where the type of [] cannot be inferred, Microsoft.Quantum.Arrays.EmptyArray can be used instead.
+    // The following code is valid.
+    let empty4 = EmptyArray<Int>();
+    Message($"Empty array: {empty4}");
+}
+```
+
+### Repeated constant arrays
+
+The semantics of `[x, size = n]` are identical to `new T[n]`, except that the initial value is now given by `x`:
+
+1. `n` may be an integer literal or a variable of integer type.
+2. If `n` is negative, a runtime error occurs.
+
+```qsharp
+// Create an array of length 10 of type Bool[] where every value is false.
+let bools = [false, size = 10];
+
+// The following examples are syntax errors:
+let wrong1 = [size = 10, false];
+let wrong2 = [false, true, size = 10];
+let wrong3 = [false, size = 10, true];
+```
+
+# Implementation
+
+The implementation of `[x, size = n]` should be a relatively straightforward extension of the current implementation for `new T[n]`.
+It will require parser and syntax tree changes, as well as changes to code generation and potentially a small amount of runtime support.
+
+The implementation for `[]` will require enhancements to type inference.
+One possible approach is to create a placeholder type variable for each usage of `[]`.
+When an expression with a placeholder type variable is used in a context that requires a specific type, this context can refine the placeholder type.
+Once all placeholders are refined, they can be replaced with their concrete types if one was determined.
+Since all specialization signatures must have explicit parameter and return types, the type inference for a placeholder variable cannot extend beyond its containing specialization.
+
+## Timeline
+
+The bulk of the work is expected to be in implementing type inference for the empty array literal.
+However, since it is limited to only the empty array literal, and bounded by the scope of a specialization, the complexity should be managable within a month or two of work.
+The remaining work of adding repeated constant array literals and deprecating the old array literals should not take much time.
+
+# Further Considerations
+
+## Related Mechanisms
+
+The syntax for allocating arrays of qubits is similar to the existing default-initialized array constructor syntax:
+
+```qsharp
+use qs = Qubit[n];
+```
+
+For consistency, it may make sense to change this syntax as well.
+The [proposal for allocatable types and generalization of initializers](https://github.com/microsoft/qsharp-language/pull/41) is addressing this.
+
+## Impact on Existing Mechanisms
+
+The [`Default`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.core.default) function depends on default-initialized array constructors, and it will also be deprecated.
+
+## Anticipated Interactions with Future Modifications
```

Comment:
It would be good to add a paragraph on both multidimensional arrays (that is actually quite an important factor), and also the possibility of array comprehension.